src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -57,6 +57,93 @@ void TestNetwork(ModelType& model,
   REQUIRE(classificationError <= classificationErrorThreshold);
 }
 
+// network1 should be allocated with `new`, and trained on some data.
+template<typename MatType = arma::mat, typename ModelType>
+void CheckCopyFunction(ModelType* network1,
+                       MatType& trainData,
+                       MatType& trainLabels,
+                       MatType& testData,
+                       MatType& testLabels,
+                       const size_t maxEpochs,
+                       const double classificationErrorThreshold)
+{
+  ens::RMSProp opt(0.01, 32, 0.88, 1e-8, maxEpochs * trainData.n_cols, -1);
+  network1->Train(trainData, trainLabels, opt);
+
+  MatType predictionTemp;
+  network1->Predict(testData, predictionTemp);
+  MatType prediction = arma::zeros<MatType>(1, predictionTemp.n_cols);
+
+  for (size_t i = 0; i < predictionTemp.n_cols; ++i)
+  {
+    prediction(i) = arma::as_scalar(arma::find(
+        arma::max(predictionTemp.col(i)) == predictionTemp.col(i), 1)) + 1;
+  }
+
+  FFN<> network2(*network1);
```

Comment:
To be really careful, I think we should also test the copy operator, like this:

```
FFN<> network3;
network3 = *network1;
```

then we can do the same checks to `network3.Predict()` as we are doing for `network2`. :+1:

Reply:
Done